raiden-ts/src/services/epics.ts

```diff
@@ -695,3 +706,146 @@ export const monitorRequestEpic = (
       ),
     ),
   );
+
+export const udcWithdrawRequestEpic = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  { userDepositContract, address, log, signer, config$, provider }: RaidenEpicDeps,
+): Observable<udcWithdrawPlanned | udcWithdraw.failure> => {
+  return action$.pipe(
+    filter(udcWithdraw.request.is),
+    withLatestFrom(from(userDepositContract.functions.balances(address))),
+    concatMap(([action, balance]) => {
+      const amount = action.payload.amount;
+      try {
+        assert(amount.gt(Zero), [
+          ErrorCodes.UDC_PLAN_WITHDRAW_GT_ZERO,
+          {
+            amount: amount.toString(),
+          },
+        ]);
+
+        assert(balance.sub(amount).gte(Zero), [
+          ErrorCodes.UDC_PLAN_WITHDRAW_EXCEEDS_AVAILABLE,
+          {
+            balance: balance.toString(),
+            amount: amount.toString(),
+          },
+        ]);
+      } catch (e) {
+        return of(udcWithdraw.failure(e, action.meta));
+      }
+      const contract = getContractWithSigner(userDepositContract, signer);
+      return from(contract.functions.planWithdraw(amount)).pipe(
+        assertTx('planWithdraw', ErrorCodes.UDC_PLAN_WITHDRAW_FAILED, { log }),
+        concatMap((txHash) =>
+          combineLatest([
+            from(provider.getTransactionReceipt(txHash)),
+            state$.pipe(pluckDistinct('blockNumber')),
+            config$.pipe(pluckDistinct('confirmationBlocks')),
+          ]).pipe(
+            first(
+              ([receipt, blockNumber, confirmationBlocks]) =>
+                !!receipt.blockNumber && receipt.blockNumber + confirmationBlocks <= blockNumber,
+            ),
+            map(() => txHash),
+          ),
+        ),
+        concatMap((txHash) => {
+          return from(userDepositContract.functions.withdraw_plans(address)).pipe(
+            map(({ amount, withdraw_block }) =>
+              udcWithdrawPlanned(
+                {
+                  amount: amount as UInt<32>,
+                  block: withdraw_block as UInt<32>,
+                },
+                {
+                  txHash,
+                },
+              ),
+            ),
+          );
+        }),
+        catchError((err) => {
+          log.error('Planning udc withdraw failed', err);
+          return of(udcWithdraw.failure(err, action.meta));
+        }),
+      );
+    }),
+  );
+};
+
+export const udcCheckWithdrawPlannedEpic = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+  { userDepositContract, address }: RaidenEpicDeps,
+): Observable<udcWithdrawPlanned> => {
+  return action$.pipe(
+    filter(newBlock.is),
+    first(),
+    concatMap(() =>
+      from(userDepositContract.functions.withdraw_plans(address)).pipe(
```

Comment:
As you're not using newBlock, and do want to check this on subscription, may I suggest you put `defer(() => userDepositContract.functions.wihtdraw_plans(address)).pipe(...` actually as the top-level observable? No need for action$ here, but `defer` is important to call the promise only at subscription.